### PR TITLE
Fixed #467: allowed multiple prop replace syntax

### LINKED ISSUE #467: Allow same prop replace syntax in props and ml-config
In properties files you use ${app-name}, in ml-config you use @ml.app-name. The difference is confusing to my opinion. And not adding { } in ml-config makes people believe @ml.app-name-rest-modules is a full property name, where it actually is just a partial one (@ml.app-name + '-rest-modules').

I vote for adding different syntax to ml-config as well, preferably ${propname} like in properties themselves, alternatively @{ml.propname}. Thoughts?

### PR DESCRIPTION
**Pull request by grtjn**
Fixes #467

### CODE REVIEW

**Comment by grtjn** [comment]
I didn't replace all placeholders to distinguish between real properties, and inserts of pieces of config. Only exception is ssl cert stuff. I replaced that in a different PR by an insert as well.

**Comment by grtjn** [comment]
Note to self: consider including support for #414..

**Comment by grtjn** [comment]
I actually did pile up #414, since it was just a one-line change..

File: deploy/lib/server_config.rb [hunk 1]
```diff
@@ -112,7 +112,7 @@ def info
       logger.info "<info>"
       logger.info "\s\s<isJar>#{@@is_jar}</isJar>"
       logger.info "\s\s<properties>"
-      @properties.sort {|x,y| y <=> x}.each do |k, v|
+      @properties.sort {|x,y| x <=> y}.each do |k, v|
```
Comment (dmcassel) [inline deploy/lib/server_config.rb]: If I remember right, the reason we put this sort in place is to make sure we replace the longest properties first. That way, if we see "@ml.app-name-foo-bar" and we have properties "app-name" and "app-name-foo", we want to use "app-name-foo" for the substitution. 

What's the reason for changing the sort order?

File: deploy/lib/server_config.rb [hunk 2]
```diff
@@ -2172,12 +2164,32 @@ def build_config(config_file)
...
+    # escape unresolved braces, they have special meaning in XQuery
+    config.gsub!("{", "{{")
+    config.gsub!("}", "}}")
 
     config
   end
+  
+  def replace_properties(contents, name)
+    @properties.sort {|x,y| y <=> x}.each do |k, v|
+      # new property syntax
```
Comment (dmcassel) [inline deploy/lib/server_config.rb]: Adding a comment showing the new property syntax will be helpful down the road.

File: deploy/lib/server_config.rb [hunk 3]
```diff
@@ -2172,12 +2164,32 @@ def build_config(config_file)
...
+      n = k.sub("ml.", "")
+      contents.gsub!("@{#{n}}", v)
+      contents.gsub!("${#{n}}", v)
+      
+      # backwards compat
```
Comment (dmcassel) [inline deploy/lib/server_config.rb]: Adding a comment showing the old property syntax will be helpful down the road.

**Comment by dmcassel** [comment]
Please rebase

File: deploy/lib/server_config.rb (hunk 1 above)
Comment (grtjn) [inline deploy/lib/server_config.rb]: You are correct. Descending order is necessary for replacement, but these lines concern display/output only. Ascending order reads much easier..

I'll rebase and update the comments..

**Comment by grtjn** [comment]
Better like this?

**Pull request merged**